src/openforms/contrib/objects_api/clients/objects.py

```diff
@@ -4,14 +4,21 @@
 
 
 class ObjectsClient(NLXClient):
-    def create_object(self, objecttype_url: str, record_data: dict) -> dict:
-
+    def create_or_update_object(
```

Comment:
I had some doubts if I should add a new method here `update_object` but I realized that it's almost the same code, so feel free to give an opinion.

Reply:
I think it's best to have the client classes not know about specific logic and flows and be as close to the API spec in some way, and instead the logic should leave in the caller's code:

```python
if initial_data_reference:
    client.update_object(...)
else:
    client.create_object(...)
```